client/web/src/components/externalServices/AddExternalServicesPage.tsx

```diff
@@ -99,6 +101,8 @@ export const AddExternalServicesPage: FC<AddExternalServicesPageProps> = ({
         ]
     }
 
+    const servicesByGroup = computeExternalServicesGroup(codeHostExternalServices, allowedCodeHosts)
```

Comment:
Recomputing the grouping on every render seems super inefficient. Can be improved by using `useMemo`:
```suggestion
    const servicesByGroup = useMemo(() => (
        computeExternalServicesGroup(codeHostExternalServices, allowedCodeHosts)
    ), [codeHostExternalServivecs, allowedCodeHosts])
```